cmake/modules/configuration_files.cmake

```diff
@@ -77,8 +77,15 @@ zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR}/boards DTS APP_BOARD_DTS SUFFIX
 
 zephyr_get(DTC_OVERLAY_FILE SYSBUILD LOCAL)
 if(NOT DEFINED DTC_OVERLAY_FILE)
-  zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR} DTS DTC_OVERLAY_FILE
-              NAMES "${APP_BOARD_DTS};${BOARD}.overlay;app.overlay" SUFFIX ${FILE_SUFFIX})
+  foreach(file ${APP_BOARD_DTS})
+    zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR} DTS DTC_OVERLAY_FILE
+                NAMES "${file}")
+  endforeach()
```

Comment:
`zephyr_file()` call here: https://github.com/zephyrproject-rtos/zephyr/blob/719b3380555e68843efa54073a256dd62c231751/cmake/modules/configuration_files.cmake#L76
has already checked for existence of files as well as created a list of files when revision is used.

Thus there is no need for an extra `zephyr_file()` call for each file, just check if `APP_BOARD_DTS` is defined, like this:
```
if(NOT DEFINED DTC_OVERLAY_FILE)
  if(DEFINED APP_BOARD_DTS)
    set(DTC_OVERLAY_FILE ${APP_BOARD_DTS})
  else()
     zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR} DTS DTC_OVERLAY_FILE
                NAMES "${BOARD}.overlay;app.overlay" SUFFIX ${FILE_SUFFIX})
  endif()
endif()
```

You can even move the code in L76 inside the if-statement, like this:
```
if(NOT DEFINED DTC_OVERLAY_FILE)
  zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR}/boards DTS DTC_OVERLAY_FILE SUFFIX ${FILE_SUFFIX})

  if(NOT DEFINED DTC_OVERLAY_FILE)
     zephyr_file(CONF_FILES ${APPLICATION_CONFIG_DIR} DTS DTC_OVERLAY_FILE
                 NAMES "${BOARD}.overlay;app.overlay" SUFFIX ${FILE_SUFFIX})
  endif()
endif()
```
It's current location probably made sense a long time ago, but not anymore afaict.

Reply:
Fixed